example/convolution1D/src/convolution1D.cpp

```diff
@@ -0,0 +1,200 @@
+/* Copyright 2023 Benjamin Worpitz, Erik Zenker, Bernhard Manfred Gruber, Jan Stephan
+ * SPDX-License-Identifier: ISC
+ */
+
+#include <alpaka/alpaka.hpp>
+#include <alpaka/example/ExampleDefaultAcc.hpp>
+
+#include <cmath>
+#include <iomanip>
+#include <iostream>
+#include <limits>
+#include <type_traits>
+//! Convolution Example
+//!
+//! 1D convolution example: Creates two 1D arrays, calculates the convolution integral using those arrays.
+//! Array sizes are hardcoded.
+//!
+
+// Size of 1D arrays to be used in convolution integral
+// In signal processing domain, the term "kernel" is used for the matrix or array used in convolution integral.
+// Here instead of "kernel" the term "filter" is used because kernel has a different meaning in GPU programming.
+constexpr size_t FilterSize = 3;
+constexpr size_t InputSize = 8;
```

Comment:
I this example, it will not cause any problems, but I would avoid to use globals where it is possible. Most of the people simply copies the code and don't think about bad practice.

I suggest to move it in the begin of the `main()`.

Reply:
done, thanks.